spec/frame.test.js

```diff
@@ -0,0 +1,41 @@
+var Frame = require('../public/src/frame')
+
+describe('Frame Class', function () {
+  describe('#addScore', function () {
+    var frame = new Frame()
+
+    beforeEach(function () {
+      frame.scores = []
+    })
+
+    it('inserts score into scores array', function () {
+      frame.addScore(3)
+      expect(frame.scores[0]).toEqual(3)
+    })
+
+    it('only accepts three scores', function () {
+      frame.addScore(7)
+      frame.addScore(3)
+      frame.addScore(4)
+      frame.addScore(1)
```

Comment:
I would expect the result of trying to add more than 3 scores would be an exception, rather than silent failure. When we do invalid or unexpected things in code, we want to make it obvious to the code that is calling that this is invalid. I think this applies to a few other bits of your code as well - when you validate, through an exception or make it obvious in some other way that the thing that has been attempted has failed, and let the code that is calling the method deal with it (usually by catching the exception and outputting some sort of error message to the user to a log file).